homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,41 @@
+"""Support for Modbus lights."""
+import logging
+from typing import Optional
+
+from homeassistant.components.light import LightEntity
+from homeassistant.const import CONF_LIGHTS, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusSwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+):
+    """Read configuration and create Modbus lights."""
+    if discovery_info is None:
+        _LOGGER.warning(
```

Comment:
This is a new platform. Do we need this warning?

Reply:
I added this as a safeguard. If users configure the platform according to docs, this case won't happen. I can remove this warning if you prefer.

Reply:
I don't think we need the warning for the new platforms.

Reply:
OK, I'll remove it then.